packages/flutter_tools/lib/src/asset.dart

```diff
@@ -317,7 +319,9 @@ DevFSContent _createFontManifest(Map<String, dynamic> manifestDescriptor,
     fonts.addAll(manifestDescriptor['fonts']);
   if (fonts.isEmpty)
     return null;
-  return new DevFSStringContent(JSON.encode(fonts));
+  String json = JSON.encode(fonts);
+  json = json.replaceAll(_assetJsonLeadingParentRegExp, '"asset":"');
```

Comment:
I wouldn't try to make a general converter.  I'd just handle the specific schema we expect for fonts.